tests/test-team/template-mgmt-component-tests/email/template-mgmt-preview-email-page.component.ts

```diff
@@ -0,0 +1,155 @@
+import { test, expect } from '@playwright/test';
+import SessionStorageHelper from '../../helpers/session-storage-helper';
+import { TemplateMgmtPreviewEmailPage } from '../../pages/email/template-mgmt-preview-email-page';
+import { SessionFactory } from '../../helpers/session-factory';
+import {
+  assertGoBackLink,
+  assertLoginLink,
+  assertNotifyBannerLink,
+  assertSkipToMainContent,
+} from '../template-mgmt-common.steps';
+
+const sessions = {
+  empty: SessionFactory.createEmailSession('empty-email-preview-session'),
+  valid: {
+    ...SessionFactory.createEmailSession('valid-email-preview-session'),
+    emailTemplateName: 'test-template-email',
+    emailTemplateSubjectLine: 'test-template-subject-line',
+    emailTemplateMessage: 'test-template-message',
+  },
+};
+
+test.describe('Preview Email message template Page', () => {
+  const sessionStorageHelper = new SessionStorageHelper(
+    Object.values(sessions)
+  );
+
+  test.beforeAll(async () => {
+    await sessionStorageHelper.seedSessionData();
+  });
+
+  test.afterAll(async () => {
+    await sessionStorageHelper.deleteSessionData();
+  });
+
+  test('when user visits page, then page is loaded', async ({
+    page,
+    baseURL,
+  }) => {
+    const previewEmailTemplatePage = new TemplateMgmtPreviewEmailPage(page);
+
+    await previewEmailTemplatePage.loadPage(sessions.valid.id);
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/preview-email-template/${sessions.valid.id}`
+    );
+
+    // Note; I've broken this check into 2 pieces otherwise it's an unreadable mess of;
```

Comment:
you can chain assertions if you want to include it in the same check

Reply:
I like the idea but I can't chain on a promise, unless you've got a suggestion?

Reply:
With a promise the only chains you have is; .then, .catch or .finally. 

Promise(fn()).then((result) => expect(something)).